src/super_gradients/training/models/segmentation_models/segformer.py

```diff
@@ -0,0 +1,599 @@
+import math
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from super_gradients.training.models import HpmStruct
+from super_gradients.training.utils import get_param
+from super_gradients.training.models.segmentation_models.segmentation_module import SegmentationModule
+from super_gradients.common.abstractions.abstract_logger import get_logger
+
+"""
+paper:  SegFormer: Simple and Efficient Design for Semantic Segmentation with Transformers
+        ( https://arxiv.org/pdf/2105.15203.pdf )
+code adopted from git repo: https://github.com/sithu31296/semantic-segmentation
+
+Imagenet-1k pre-trained backbone weights taken and adapted from: https://github.com/sithu31296/semantic-segmentation
+
+"""
+
+logger = get_logger(__name__)
+
+
+# TODO: this function (and trunc_normal_) are copy-pasted from BEIT model code. We need to consider implementing
+#  it in a more general location
+def _no_grad_trunc_normal_(tensor, mean, std, a, b):
+    # Cut & paste from PyTorch official master until it's in a few official releases - RW
+    # Method based on https://people.sc.fsu.edu/~jburkardt/presentations/truncated_normal.pdf
+    def norm_cdf(x):
+        # Computes standard normal cumulative distribution function
+        return (1.0 + math.erf(x / math.sqrt(2.0))) / 2.0
+
+    if (mean < a - 2 * std) or (mean > b + 2 * std):
+        logger.warn("mean is more than 2 std from [a, b] in nn.init.trunc_normal_. " "The distribution of values may be incorrect.", stacklevel=2)
+
+    with torch.no_grad():
+        # Values are generated by using a truncated uniform distribution and
+        # then using the inverse CDF for the normal distribution.
+        # Get upper and lower cdf values
+        lower = norm_cdf((a - mean) / std)
+        upper = norm_cdf((b - mean) / std)
+
+        # Uniformly fill tensor with values from [l, u], then translate to
+        # [2l-1, 2u-1].
+        tensor.uniform_(2 * lower - 1, 2 * upper - 1)
+
+        # Use inverse cdf transform for normal distribution to get truncated
+        # standard normal
+        tensor.erfinv_()
+
+        # Transform to proper mean, std
+        tensor.mul_(std * math.sqrt(2.0))
+        tensor.add_(mean)
+
+        # Clamp to ensure it's in the proper range
+        tensor.clamp_(min=a, max=b)
+        return tensor
+
+
+def trunc_normal_(tensor, mean=0.0, std=1.0, a=-2.0, b=2.0):
```

Comment:
Please consider using https://pytorch.org/docs/stable/nn.init.html#torch.nn.init.trunc_normal_ instead as it's probably same implementation, but library method is preferrable.